crates/oxc_linter/src/rules/eslint/no_restricted_imports.rs

```diff
@@ -273,6 +345,30 @@ impl RestrictedPattern {
 
         GlobResult::Found
     }
+
+    fn get_regex_result(&self, name: &NameSpan) -> bool {
+        let Some(regex) = &self.regex else {
+            return false;
+        };
+
+        regex.find(name.name()).is_some()
```

Comment:
you can call `is_match` in `regex` which [according to the lib](https://docs.rs/regex/latest/regex/struct.Regex.html#method.is_match) allows us to do slightly less work. we can also chain this into one expression which is mainly just stylistic 😄 (same comments for the other 2 functions here)

```suggestion
        self.regex.as_ref().is_some_and(|regex| regex.is_match(name.name()))
```